lib/httpserver/httpserver.go

```diff
@@ -687,3 +694,26 @@ func GetRequestURI(r *http.Request) string {
 	}
 	return requestURI + delimiter + queryArgs
 }
+
+func collectCipherSuites(definedCipherSuites []string) []uint16 {
+	var cipherSuites []uint16
+
+	supportedCipherSuites := tls.CipherSuites()
+	for _, scf := range supportedCipherSuites {
+		for _, gotSuite := range definedCipherSuites {
+			if strings.EqualFold(scf.Name, gotSuite) {
+				cipherSuites = append(cipherSuites, scf.ID)
+			} else {
+				parseUint, err := strconv.ParseUint(strings.TrimLeft(gotSuite, "0x"), 16, 16)
```

Comment:
For me, it looks like, uint parsing for cipher suites is not needed. Just use cipher name and return and error, if provided cipher name is not supported by server.

  uint number is suitable only for golang developers. Common users use cipher name.

Reply:
updated logic